aqa_2/src/tests/task2.test.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  loginPageSelectors_karpovich as loginPageSelectors,
+  registrationPageSelectors_karpovich as registrationPageSelectors,
+} from "../data/locators";
+import { loginMessages_karpovich as loginMessages } from "../data/messages";
+import { credentials_karpovich as credentials } from "../data/credentials";
+
+describe("[UI] Login Smoke tests", () => {
+  before(async function () {
+    await browser.maximizeWindow();
+  });
+
+  beforeEach(async function () {
+    await browser.url("https://anatoly-karpovich.github.io/demo-login-form/");
+  });
+
+  afterEach(async function () {
+    await browser.execute("window.localStorage.clear()");
+  });
+
+  // -------------- REGISTRATION -------------- //
+  describe("Registration Page", () => {
+    beforeEach(async function () {
+      await $(loginPageSelectors.registerButton).click();
+    });
+
+    it("Should register with valid credentials", async function () {
+      for (const cred of credentials.validCredentials) {
+        it(`${(cred.username, cred.password)}`, async () => {
+          await registerOrLoginUser(
+            registrationPageSelectors.usernameInput,
+            registrationPageSelectors.passwordInput,
+            registrationPageSelectors.registerButton,
+            cred.username,
+            cred.password
+          );
+
+          await checkSelectorTextWithContain(
+            registrationPageSelectors.errorMessage,
+            loginMessages.registrationSuccess
+          );
+        });
+      }
+    });
```

Comment:
Чтоб не создавать второй `it` блок
```suggestion   
   for (const { username, password } of credentials.validCredentials) {
      it(`Should register with valid credentials login: ${username}, password: ${password}`, async function () {
          await registerOrLoginUser(
            registrationPageSelectors.usernameInput,
            registrationPageSelectors.passwordInput,
            registrationPageSelectors.registerButton,
            username,
            password
          );

          await checkSelectorTextWithContain(
            registrationPageSelectors.errorMessage,
            loginMessages.registrationSuccess
          );
        });
    });
  }
```